### DISCUSSION TITLE
Planner buffer size default

### DISCUSSION THREAD

**Discussion by phil-barrett** [terjeio/grblHAL#162] [General]
Is there any downside to making the default planner buffer size larger? Say 1024? I suspect the original size of 36 was to deal with the limitations of the ATMega386P.

In a world where most, if not all, target machines have a lot more RAM, it makes sense to relax some constraints.

**Comment by terjeio** [comment]
> Is there any downside to making the default planner buffer size larger?

Performance hit? I'll have to check that but I do not think so. 1024 would require 40K+ of RAM as each entry is around 40 bytes. Some analysis should IMO be done before increasing it to find out what is an optimal value. High speed rendering of bitmaps is perhaps the only use-case wher this is relevant, and this would likely require a significant increase in the baud rate as well - at least for UART based connections?

If increased it should be done per processor as some does not have enough RAM for a large size. The size _can_ be specified as a compiler option with -D, easy to do in a real IDE via its UI but not so in a hobby IDE such as Arduino?

BTW the default size was 15 or 16 for 8-bit grbl depending on whether line numbers were used or not.
